src/test/java/teammates/test/cases/BaseTestCaseWithDatastoreAccess.java

```diff
@@ -267,6 +269,13 @@ private void equalizeIrrelevantData(StudentAttributes expected, StudentAttribute
         expected.lastName = StringHelper.splitName(expected.name)[1];
     }
 
+    /**
+     * Removes existing entities of {@code testData} in datastore,
+     * then persists entities of {@code testData} in datastore.
+     * Removing or persisting {@link AdminEmailAttributes} should only be performed in {@link GaeSimulation}
+     * and avoided in production environment, as non-testing data may be removed
+     * and new entities created may affect normal usage of the Admin Email functions
```

Comment:
Same here. Even more so because `GaeSimulation` is an API only relevant to `BaseComponentTestCase`. This class should not even know that there is such a thing as `GaeSimulation`.